packages/iris-grid/src/IrisGridTableModel.ts

```diff
@@ -78,6 +78,24 @@ class IrisGridTableModel
     return this.table.applyCustomColumns != null;
   }
 
+  getMemoizedKeyColumnSet = memoize(
+    (tableColumns: DhType.Column[], inputTableKeys: string[]) =>
+      new Set(
+        tableColumns
+          .filter((column: DhType.Column) =>
+            inputTableKeys.includes(column.name)
+          )
+          .map((column: DhType.Column) => column.name)
+      )
+  );
```

Comment:
I don't know why I suggested memoizing on `table.columns` and keys... we really only need the names of the columns here. Should just memoize on `inputTableKeys`. Also should take the array input as a `readonly` array to be safe. In tandem with my other column below, I think it should just be:

```typescript
getMemoizedKeyColumnSet = memoize((inputTableKeys?: readonly string[]) => new Set(inputTableKeys ?? EMPTY_ARRAY))
```